### PR TITLE
[PORT] Remove code/__DEFINES/misc.dm

### PR DESCRIPTION
**Pull request by mystery3525**
## About The Pull Request

Ports: https://github.com/tgstation/tgstation/pull/63879/

Yet another monster code re-structuring, this time defines!

> This removes code/__DEFINES/misc.dm and moves all the defines to either:
>
> - another existing define file
> - new define file
> - local .dm file if the define was only used in one file

> I also deleted defines that were not being used and added documentation to all of the ones that were moved out of misc.dm

I also moved code/__DEFINES/research/anomalies.dm out of it's snowflake folder that it had for no apparent reason

## Why It's Good For The Game

> Why was this needed? People were basically using the misc.dm file as a dumpster to toss all their defines into that was creating one giant mess. The defines have been organized into their proper groups and files now

This also follows our code standards much more better with more specific files.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

it compiles 🤯

</details>

## Changelog
:cl: timothymtorres ported by mystery3525
code: Move defines to their local .dm files
code: Remove code/__DEFINES/misc.dm
code: Moved code/__DEFINES/research/anomalies.dm
refactor: Adds some documentation to defines
/:cl:

### CODE REVIEW

File: code/__DEFINES/mobs.dm [hunk 1]
```diff
@@ -481,3 +489,103 @@
...
+/// Get the client from the var
+#define CLIENT_FROM_VAR(I) (ismob(I) ? I:client : (istype(I, /client) ? I : (istype(I, /datum/mind) ? I:current?:client : null)))
+
+/// The mob will vomit a green color
+#define VOMIT_TOXIC 1
+/// The mob will vomit a purple color
+#define VOMIT_PURPLE 2
+
+/// Messages when (something) lays an egg
+#define EGG_LAYING_MESSAGES list("lays an egg.","squats down and croons.","begins making a huge racket.","begins clucking raucously.")
```
Comment (itsmeow) [inline code/__DEFINES/mobs.dm:591]: this is the best thing ive seen all day, just wanted to share that

File: code/modules/events/pirates.dm [hunk 2]
```diff
@@ -1,5 +1,11 @@
 GLOBAL_VAR_INIT(pirates_spawned, FALSE)
 
+// Pirates threat
+/// No way
+#define PIRATE_RESPONSE_NO_PAY "pirate_answer_no_pay"
+/// We'll pay
+#define PIRATE_RESPONSE_PAY "pirate_answer_pay"
```
Comment (itsmeow) [inline code/modules/events/pirates.dm:7]: I could've sworn I needed this in another file, ah well

**Review by itsmeow — APPROVED** [review]
incredibly based

literally moves code around and fixes some bad spelling

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -5,6 +5,12 @@
  * 		Recipe list datum
  */
 
+//stack recipe placement check types config
+/// checks if there is an object of the result type in any of the cardinal directions
+#define STACK_CHECK_CARDINALS "cardinals"
+/// checks if there is an object of the result type within one tile
+#define STACK_CHECK_ADJACENT "adjacent"
```
Comment (itsmeow) [inline code/game/objects/items/stacks/stack.dm:12]: these don't appear to be undefined

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by itsmeow — APPROVED** [review]

**Comment by mystery3525** [comment]
bruh rerun the tests?

**Pull request merged**